### PR TITLE
✨🔀 Gegenüberstellung NWK / Praktikumsstellen

### LINKED ISSUE #19: Gegenüberstellung von NWKs und Praktikumsplätzen
Als örtl. Ausbildungsleitung möchte ich eine kombinierte Übersicht von NWKs und Praktikumsplätzen haben, sodass ich an einem Ort die zuzuordnenden NWKs und die Praktikumsplätze sehen kann.

**Akzeptanzkriterien:**
- [x] Die aktiven NWKs sollen in einer Liste im linken Bildschirmteil angeordnet sein
- [x] Die Praktikumsplätze sollen in einer Liste im rechten Bildschirmteil angeordnet sein
- [x] Die Praktikumsplätze sind nach den zugehörigen Abteilungen gruppiert
- [x] Zur Zuweisung sind nur Stellen aus dem letzten Meldezeitraum verfügbar

Fragen für Refinement:
- [x] Nach was wird standardmäßig sortiert?

Zusätzliche Infos zur Implementierung:
Erste Design-Idee:  
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBZ0xQIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--8fdfac544158a5b64852d31f68e6e5468d723975/grafik.png)

Diese Infos sollen zur Praktikumsstelle angezeigt werden in der Übersicht:
- Abteilung
- Studiengang/Ausbildungsrichtung
- Studien-/Ausbildungsjahr
- wenn vorhanden Namentliche Anforderung

"aktiven": siehe https://app.zenhub.com/workspaces/praktikumsplaner-650bff5c552a4616fb4b3a8d/issues/gh/it-at-m/praktikumsplaner/12

---
**Definition of Ready (DoR):**

- [x] Die User Story ist klar und verständlich aus der Perspektive der betroffenen Persona formuliert.
- [x] Vollständige und klare Akzeptanzkriterien sind definiert.
- [x] Die Aufwandsschätzung für das Issue wurde durchgeführt und dokumentiert.

---
**Definition of Done (DoD):**

- [x] Unittest gepflegt
- [x] Dokumentation ist gepflegt
- [ ] Usability Test wurden durchgeführt
- [ ] Auf Testumgebung verfügbar
- [ ] Akzeptanzkriterien sind alle erfüllt

### PR DESCRIPTION
**Pull request by danielnwz**
**Description:**  

This feature is the preparatory work for assigning NWKs to Praktikumsstellen. NWKs and Praktikumsstellen are compared here so that they can find each other quickly and easily. All NWKs that are newly created are active for the time being and remain so forever. 
Only all Praktikumsstellen that were reported in the last reporting period, which is not the current active one, are displayed. 
In addition, when reporting an Praktikumsstelle, each Dienststelle is now shortened by spaces, hyphens, "ITM", "IT@M" and "RIT" in order to better display the grouping of the Praktikumsstellen. 

The design is currently only a WIP design, as the later design is not yet available.

**Reference:**

closes #19 

Notifying additional team members:

@mirrodi @MrSebastian @AnHo314

---
**Definition of Done (DoD):**

- [x] Unittests gepflegt
- [x] Dokumentation ist gepflegt

### CODE REVIEW

File: praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/service/PraktikumsstellenService.java [hunk 1]
```diff
@@ -20,13 +31,69 @@ public class PraktikumsstellenService {
...
+
+        return abteilungsMap;
+    }
+
+    private String getHauptabteilung(String dienststelle) {
+        int index = -1;
+        for (int i = 0; i < dienststelle.length(); i++) {
+            if (Character.isDigit(dienststelle.charAt(i))) {
+                index = i;
+                break;
```
Comment (Tuckerle) [inline praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/service/PraktikumsstellenService.java:90]: I think a return would be better to understand what happens and then return dienststelle by default if the loop did not trigger the return

File: praktikumsplaner-frontend/frontend/src/components/ActiveNWKList.vue [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { ref, onMounted } from "vue";
+import NWK from "@/types/NWK";
+import NWKService from "@/api/NWKService";
+
+const nwks = ref<NWK[]>([]);
+
+onMounted(() => {
+    refreshTasks();
+});
+function refreshTasks() {
```
Comment (Tuckerle) [inline praktikumsplaner-frontend/frontend/src/components/ActiveNWKList.vue:28]: i think this name is wrong/not good

File: praktikumsplaner-frontend/frontend/src/components/PraktikumsstellenList.vue [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import { ref, onMounted } from "vue";
+import Praktikumsstelle from "@/types/Praktikumsstelle";
+import PraktikumsstellenService from "@/api/PraktikumsstellenService";
+
+const praktikumsstellen = ref<Map<string, Praktikumsstelle[]>>();
+
+onMounted(() => {
+    refreshTasks();
+});
+function refreshTasks() {
```
Comment (Tuckerle) [inline praktikumsplaner-frontend/frontend/src/components/PraktikumsstellenList.vue:89]: i think this name is wrong/not good

**Review by LuisEinig — CHANGES_REQUESTED** [review]

File: praktikumsplaner-frontend/frontend/src/router.ts [hunk 4]
```diff
@@ -71,6 +72,11 @@ export default new Router({
             name: "SuperNWK",
             component: MeldungSuperNwk,
         },
+        {
+            path: "/gegenueberstellung",
+            name: "Gegenueberstellung",
+            component: gegenUeberstellung,
```
Comment (LuisEinig) [inline praktikumsplaner-frontend/frontend/src/router.ts:78]: U should be lowercase

**Review by LuisEinig — APPROVED** [review]

**Pull request merged**